### DISCUSSION TITLE
What is the recommended approach to both having a default layout and having pages that don't use that layout?

### DISCUSSION THREAD

**Discussion by zkdiff** [remix-run/remix#5106] [Q&A] [answered]
https://remix.run/docs/en/v1/guides/routing#rendering-route-layout-hierarchies

In the documentation linked above, the example is given that the root.jsx file renders the sidebar and then an outlet for nested routes. This is exactly what I want for the majority of the application, however there are a few pages which should not include the application shell as rendered by root.jsx.

Sometimes, one or more of these "non-appshelled" pages share another common layout.

What is the recommended filesystem structure to model this situation? Is it possible or advisable to somehow create multiple root.jsx files?

I could solve this by simply including an extra route path for the majority of the application, for example:
```
app
├── root.jsx
└── routes
   ├── some-landing-page.jsx *
   ├── some-terms-page.jsx *
   ├── some-other-page.jsx *
   ├── app.jsx
   └── app
        ├── accounts.jsx
        ├── dashboard.jsx
        ├── expenses.jsx
        ├── index.jsx
        ├── reports.jsx
        ├── sales
        │   ├── customers.jsx
        │   ├── deposits.jsx
        │   ├── index.jsx
        │   ├── invoices
        │   │   ├── $invoiceId.jsx
        │   │   └── index.jsx
        │   ├── invoices.jsx
        │   └── subscriptions.jsx
        └── sales.jsx
```

What is the way to implement this without adding the extra `app` route segment when all of the asterisk'd routes need to have a different root layout from the rest of the application?

**Comment by sergiodxa** [comment, marked as answer]
What you need is a pathless layout route, so you can have a parent route adding UI without adding path segments.

**Reply by zkdiff** [reply]
I guess that is the part I'm having trouble understanding. What would that look like for root?

**Reply by zkdiff** [reply]
Ok, so I see that it is actually pretty simple!
A directory structure like the following one works just fine and allows for an arbitrary number of so-called "root" layouts. One must simply take care to avoid any duplicate routes since you can create an index.tsx in each __folder whereas usually the filesystem would prevent such conflicts.

```
app/
|-- entry.client.tsx
|-- entry.server.tsx
|-- root.tsx
|-- routes
|   |-- __auth
|   |   |-- sign-in.tsx
|   |   |-- sign-out.tsx
|   |   `-- sign-up.tsx
|   |-- __restofstuff
|   |   `-- index.tsx
|   `-- __restofstuff.tsx
```

Thank you for pointing me back in the right direction, @sergiodxa. I'd looked at pathless layout routes previously but hadn't full appreciated how they worked.

**Reply by zkdiff** [reply]
And finally I have found `flat-routes` which is basically I really wanted to do in the first place:  https://github.com/remix-run/remix/discussions/4482
